github/github.go

```diff
@@ -0,0 +1,663 @@
+			return false, nil
+		default:
+			return false, fmt.Errorf("unknown status: %s", status)
+		}
+	}
+	return true, nil
+}
+
+// Wait for a PR to move into Pending.  This is useful because the request to test a PR again
+// is asynchronous with the PR actually moving into a pending state
+// TODO: add a timeout
+func (config *GithubConfig) WaitForPending(prNumber int) error {
+	for {
+		status, err := config.GetStatus(prNumber, []string{})
+		if err != nil {
+			return err
+		}
+		if status == "pending" {
+			return nil
+		}
+		glog.V(4).Info("PR is not pending, waiting for 30 seconds")
+		time.Sleep(30 * time.Second)
+	}
+}
+
+func (config *GithubConfig) GetCommits(prNum int) ([]github.RepositoryCommit, error) {
+	//TODO: this should handle paging, I believe....
+	commits, _, err := config.client.PullRequests.ListCommits(config.Org, config.Project, prNum, &github.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return commits, nil
+}
+
+func (config *GithubConfig) GetFilledCommits(prNum int) ([]github.RepositoryCommit, error) {
+	commits, err := config.GetCommits(prNum)
+	if err != nil {
+		return nil, err
+	}
+	filledCommits := []github.RepositoryCommit{}
+	for _, c := range commits {
+		commit, _, err := config.client.Repositories.GetCommit(config.Org, config.Project, *c.SHA)
+		if err != nil {
+			glog.Errorf("Can't load commit %s %s %s", config.Org, config.Project, *commit.SHA)
+			continue
+		}
+		filledCommits = append(filledCommits, *commit)
+	}
+	return filledCommits, nil
+}
+
+func (config *GithubConfig) GetPR(prNum int) (*github.PullRequest, error) {
+	pr, _, err := config.client.PullRequests.Get(config.Org, config.Project, prNum)
+	if err != nil {
+		glog.Errorf("Error getting PR# %d: %v", prNum, err)
+		return nil, err
+	}
+	return pr, nil
+}
+
+func (config *GithubConfig) AssignPR(prNum int, owner string) error {
+	assignee := &github.IssueRequest{Assignee: &owner}
+	if config.DryRun {
+		glog.Infof("Would have assigned PR# %d  to %v but --dry-run was set", prNum, owner)
+		return nil
+	}
+	if _, _, err := config.client.Issues.Edit(config.Org, config.Project, prNum, assignee); err != nil {
+		glog.Errorf("Error assigning issue# %d to %v: %v", prNum, owner, err)
+		return err
+	}
+	return nil
+}
+
+func (config *GithubConfig) ClosePR(pr *github.PullRequest) error {
+	if config.DryRun {
+		glog.Infof("Would have editted PR# %d but --dry-run was set")
+		return nil
+	}
+	state := "closed"
+	pr.State = &state
+	if _, _, err := config.client.PullRequests.Edit(config.Org, config.Project, *pr.Number, pr); err != nil {
+		glog.Errorf("Failed to close pr %d: %v", *pr.Number, err)
+		return err
+	}
+	return nil
+}
+
+func (config *GithubConfig) OpenPR(pr *github.PullRequest) error {
+	if config.DryRun {
+		glog.Infof("Would have editted PR# %d but --dry-run was set")
+		return nil
+	}
+	var err error
+	state := "open"
+	pr.State = &state
+	// Try pretty hard to re-open, since it's pretty bad if we accidentally leave a PR closed
```

Comment:
I decided to parameterize. If anyone else wants open, I expect them to REALLY want open....